### ISSUE TITLE
Connection to redis-trib cluster failes

### ISSUE THREAD

**Issue by rwindegger** [NodeBB/NodeBB#6201]
- **NodeBB version:** 1.7.2
- **Database type:** redis [cluster using redis-trib](https://redis.io/topics/cluster-tutorial)
- **Database version:** 4.0.6

- **Exact steps to cause this issue:**
Run ./nodebb upgrade

- **What you expected:**
Successful Database deployment

- **What happened instead:**

```root@67217be1ef6b:/usr/src/app# ./nodebb upgrade

Updating NodeBB...

1. Updating package.json file with defaults...  OK

2. Bringing base dependencies up to date...
up to date in 4.986s
  OK

3. Checking installed plugins for updates...  OK
4. Updating NodeBB data store schema...
Parsing upgrade scripts...
OK | 40 script(s) found
Error occurred during upgrade

/usr/src/app/src/cli/upgrade.js:70
                        throw err;
                        ^
ReplyError: MOVED 12430 10.0.5.10:6379
    at parseError (/usr/src/app/node_modules/redis-parser/lib/parser.js:193:12)
    at parseType (/usr/src/app/node_modules/redis-parser/lib/parser.js:303:14)
```

After some research it seems that the redis library used does not support native redis clusters. From my understanding there is a drop in replacement available at: [redis-clustr](https://github.com/gosquared/redis-clustr)

Thanks for your support.

**Comment by julianlam** [comment]
Upstream discussion: https://github.com/NodeRedis/node_redis/issues/574

**tl;dr** You can use ioredis, or redis-clustr which seems to be a strong recommendation.

No objections to implementing the redis-clustr wrapper, though care should be taken to ensure that database throughput is not affected...

**Comment by julianlam** [comment]
Also @rwindegger are you available to test on a realistic-ish environment if we add this wrapper in?

**Comment by rwindegger** [comment]
@julianlam I've teared down the setup. And instead setup a mongo cluster and a single redis node for caching. I don't really have any forum running with real traffic. Just getting in touch with NodeBB. Feel free to close the issue till someone with a realistic-ish environment comes along.

**Comment by julianlam** [comment]
No worries, will close until more interest comes along.
